python_benchmarks/qr_codes/qr.py

```diff
@@ -106,60 +207,85 @@ def get_gold_corners(label_path):
-
-
-def get_norm_to_rotate_qr(gold_corner, corners, accuracy, type_dist=TypeNorm.l_inf):
-    corners = corners.reshape(-1, 4, 2)
-    dist = 1e9
-    best_id = 0
-    cur_id = 0
-    for one_corners in corners:
-        prev_dist = dist
-        dist = min(dist, get_norm(gold_corner, one_corners, type_dist))
-        for i in range(0, 3):
-            one_corners = np.roll(one_corners, 1, 0)
-            dist = min(dist, get_norm(gold_corner, one_corners, type_dist))
-        one_corners = np.flip(one_corners, 0)
-        dist = min(dist, get_norm(gold_corner, one_corners, type_dist))
-        for i in range(0, 3):
-            one_corners = np.roll(one_corners, 1, 0)
-            dist = min(dist, get_norm(gold_corner, one_corners, type_dist))
-        if dist < prev_dist:
-            best_id = cur_id
-        cur_id += 1
-
-    if type_dist is TypeNorm.intersection_over_union:
-        rect = cv.boundingRect(np.concatenate((gold_corner, corners[best_id]), dtype=np.float32))
-        mask1 = np.zeros((rect[2], rect[3]), dtype=np.uint8)
-        mask1 = cv.fillConvexPoly(mask1, gold_corner.astype(np.int32) - (rect[0], rect[1]), 255)
-        mask2 = np.zeros((rect[2], rect[3]), dtype=np.uint8)
-        mask2 = cv.fillConvexPoly(mask2, corners[best_id].astype(np.int32) - (rect[0], rect[1]), 255)
-        a = cv.countNonZero(cv.bitwise_and(mask2, mask1))
-        b = max(cv.countNonZero(cv.bitwise_or(mask2, mask1)), 1)
-        dist = 1. - a / b
-    return dist, best_id
+def get_time():
+    return datetime.datetime.now().strftime("%Y_%m_%d-%I_%M_%S_%p")
+
+
+class NumpyEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, np.ndarray):
+            return obj.tolist()
+        return json.JSONEncoder.default(self, obj)
+
+
+def set_plt():
+    # Turn interactive plotting off
+    plt.ioff()
+    plt.rcParams["figure.figsize"] = (15, 9)
+    plt.rcParams["figure.subplot.bottom"] = 0.3
+    plt.rcParams["figure.subplot.left"] = 0.05
+    plt.rcParams["figure.subplot.right"] = 0.99
+
+
+def get_and_print_category_statistic(obj_type, category, statistics, accuracy, path):
+    objs = np.array(list(deepflatten(statistics)))
+    detected = objs[objs < accuracy]
+    category_statistic = OrderedDict(
+        [("category", category), ("detected " + obj_type, len(detected) / max(1, len(objs))),
+         ("total detected " + obj_type, len(detected)), ("total " + obj_type, len(objs)),
+         ("average detected error " + obj_type, np.mean(detected))])
+    data_frame = pd.DataFrame(objs)
+    data_frame.hist(bins=500)
+    plt.title(category + ' ' + obj_type)
+    plt.xlabel('error')
+    plt.xticks(np.arange(0., float(accuracy) + .25, .25))
+    plt.ylabel('frequency')
+    # plt.show()
+    plt.savefig(path + '/' + category + '_' + obj_type + '.jpg')
+    plt.close()
+    return category_statistic
+
+
+def print_statistics(distances, accuracy, output_path, per_image_statistic):
```

Comment:
I propose to rename `accuracy` to `accuracy_threshold` everywhere, including command line parameters. It makes code analysis dramatically simpler.